packages/flutter/lib/src/widgets/nested_scroll_view.dart

```diff
@@ -0,0 +1,1002 @@
+    if (pixels != value) {
+      final double oldPixels = pixels;
+      forcePixels(value);
+      reportScrollStart();
+      reportScroll(pixels - oldPixels);
+      reportScrollEnd();
+    }
+  }
+
+  @override
+  void applyNewDimensions() {
+    _activity.applyNewDimensions();
+    owner.updateCanDrag();
+  }
+
+  void updateCanDrag(double totalExtent) {
+    state.setCanDrag(totalExtent > (viewportDimension - maxScrollExtent) || minScrollExtent != maxScrollExtent);
+  }
+
+  @override
+  void touched() {
+    owner.touched();
+  }
+
+  void propagateTouched() {
+    _activity.touched();
+  }
+
+  @override
+  Drag beginDrag(DragStartDetails details, VoidCallback dragCancelCallback) {
+    return owner.beginDrag(details, dragCancelCallback);
+  }
+
+  @override
+  void updateUserScrollDirection(ScrollDirection value) {
+    assert(false);
+    owner.updateUserScrollDirection(value);
+  }
+
+  @override
+  void reportScrollStart() {
+    _activity.dispatchScrollStartNotification(cloneMetrics(), state.notificationContext);
+  }
+
+  @override
+  void reportScroll(double delta) {
+    _activity.dispatchScrollUpdateNotification(cloneMetrics(), state.notificationContext, delta);
+  }
+
+  @override
+  void reportScrollEnd() {
+    _activity.dispatchScrollEndNotification(cloneMetrics(), state.notificationContext);
+  }
+
+  @override
+  void reportOverscroll(double value) {
+    assert(_activity.isScrolling);
+    _activity.dispatchOverscrollNotification(cloneMetrics(), state.notificationContext, value);
+  }
+
+  @override
+  void reportScrollDirection(ScrollDirection direction) {
+    new UserScrollNotification(metrics: cloneMetrics(), context: state.notificationContext, direction: direction)
+      .dispatch(state.notificationContext);
+  }
+
+  bool get shouldIgnorePointer => _activity?.shouldIgnorePointer;
+
+  void setIgnorePointer(bool value) {
+    state.setIgnorePointer(value);
+  }
+
+  @override
+  void dispose() {
+    _activity?.dispose();
+    _parent?.detach(this);
+    super.dispose();
+  }
+
+  @override
+  void debugFillDescription(List<String> description) {
+    super.debugFillDescription(description);
+    if (debugLabel != null)
+      description.add('debug label: $debugLabel');
+  }
+}
+
+class _NestedDragScrollCoordinator implements Drag {
+  // TODO(ianh): There's a lot of code duplication with DragScrollActivity here.
+  _NestedDragScrollCoordinator(
+    this.owner,
+    DragStartDetails details,
+    this.onDragCanceled,
+  ) : _lastDetails = details;
+
+  final _NestedScrollViewState owner;
+
+  final VoidCallback onDragCanceled;
+
+  dynamic get lastDetails => _lastDetails;
+  dynamic _lastDetails;
+
+  bool get _reversed {
+    assert(owner?._outerPosition?.axisDirection != null);
+    switch (owner._outerPosition.axisDirection) {
+      case AxisDirection.up:
+      case AxisDirection.left:
+        return true;
+      case AxisDirection.down:
+      case AxisDirection.right:
+        return false;
```

Comment:
Duplicated in _getRightWayUp, _reverseDirection.  Probably worth breaking out into a separate function.